ra/ra.go

```diff
@@ -1424,18 +1392,54 @@ func (ra *RegistrationAuthorityImpl) NewOrder(ctx context.Context, req *rapb.New
 		return nil, err
 	}
 
-	// TODO(#2955): Replace this with the batched methods
+	now := ra.clk.Now().UnixNano()
+	existingAuthz, err := ra.SA.GetAuthorizations(ctx, &sapb.GetAuthorizationsRequest{
+		RegistrationID: req.RegistrationID,
+		Now:            &now,
+		Domains:        parsedCSR.DNSNames,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	gotAuthzFor := make(map[string]struct{}, len(parsedCSR.DNSNames))
+	for _, v := range existingAuthz.Authz {
+		gotAuthzFor[*v.Domain] = struct{}{}
```

Comment:
I think we've been favouring using `bool` over `struct{}` in these cases for new code.